mypy/checkmember.py

```diff
@@ -40,8 +40,10 @@ def analyze_member_access(name: str,
       1. lvalue or non-lvalue access (i.e. setter or getter access)
       2. supertype access (when using super(); is_super == True and
          override_info should refer to the supertype)
+
+    actual_self is the type of E in the expression E.foo
```

Comment:
This code is pretty tricky, that's why I'm asking for a docstring (and that's probably also why it's difficult to write that docstring). I still don't quite understand how `actual_self` and `typ` are related when this gets called from `analyze_super`, and why they are different.

Maybe a better name for the argument would be `original_type`? I think that it would kind of cover both the `reported_type` and `declared_type` use case. Using `self` in the name is also a bit inconsistent with how the the type is used (at least when doing error reporting), as there might no explicit self argument in sight. 

Here's an idea for the docstring (though this only covers a part of what's going on, I think):

> Return the type of attribute `name` of `typ`. `original_type` is the most precise inferred or declared type of the base object that we have available. `typ` is generally a supertype
of `original_type`. When looking for an attribute of `typ`, we may perform recursive calls targeting the fallback type, for example. `original_type` is always the type used in the initial call.